python-checks/src/main/java/org/sonar/python/checks/cdk/S3BucketServerEncryptionCheck.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.python.tree.QualifiedExpressionImpl;
+
+@Rule(key = "S6245")
+public class S3BucketServerEncryptionCheck extends AbstractS3BucketCheck {
+
+  private static final String S3_BUCKET_UNENCRYPTED_FQN = "aws_cdk.aws_s3.BucketEncryption.UNENCRYPTED";
+  public static final String MESSAGE = "Objects in the bucket are not encrypted. Make sure it is safe here.";
+  public static final String OMITTING_MESSAGE = "Omitting 'encryption' disables server-side encryption. Make sure it is safe here.";
+
+  @Override
+  void visitBucketConstructor(SubscriptionContext ctx, CallExpression bucket) {
+    Optional<ArgumentTrace> optEncryptionType = getArgument(ctx, bucket, "encryption");
+    if (optEncryptionType.isPresent()) {
+      optEncryptionType.ifPresent(argumentTrace -> {
+        List<Expression> list = argumentTrace.trace();
+        argumentTrace.addIssueIf(isUnencrypted(list.get(list.size() - 1)), MESSAGE);
```

Comment:
You don't have to pick the last item on the list. The predicate which is tested inside `addIssueIf` will actually try to get any match. Out of the types which can be part of the trace list, only the last one can match this expression. The list only contains expressions of type NAME which are a reference to another expression or which is a value expression. The predicate can only match the value expression. Tipp: compare this to `isFalse`

Reply:
I was thinking that it may slightly reduce the overhead if there is a long trace, but in practice there are probably not too long traces. I'll remove this modification